packages/dhis2/src/tracker.js

```diff
@@ -0,0 +1,126 @@
+import { expandReferences } from '@openfn/language-common/util';
+import * as util from './Utils';
+
+/**
+ * State object
+ * @typedef {Object} Dhis2State
+ * @private
+ * @property data - The response body (as JSON)
+ * @property response - The HTTP response from the Dhis2 server (excluding the body)
+ * @property references - An array of all previous data objects used in the Job
+ */
+
+/**
+ * Options object
+ * @typedef {Object} RequestOptions
```

Comment:
I think request options should point to DHIS2 docs for each resource type instead of having `{ headers, query}` as request option.
This will make the code very tidy and precise
Eg:
```
tracker.export('trackedEntities' { orgUnits: "abc, fyc"})
```
Instead of
```
tracker.export('trackedEntities', { query: { orgUnits: "abc, fyc"}}
```
The only reason i could think of on when a user would want to parse something in headers is when they need a different output format. For example `text/csv , application/csv+gzip application/json+gzip` which could be mapped internally by providing a `parseAs` option


We should have examples for each resourceType and link to request params docs in that example